src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -209,6 +212,8 @@ private static void GetFindData(string fullPath, bool isDirectory, ref Interop.K
                 // File not found doesn't make much sense coming from a directory.
                 if (isDirectory && errorCode == Interop.Errors.ERROR_FILE_NOT_FOUND)
                     errorCode = Interop.Errors.ERROR_PATH_NOT_FOUND;
+                if (isDirectory && errorCode == Interop.Errors.ERROR_ACCESS_DENIED && ignoreAccessDenied)
+                    return;
```

Comment:
```suggestion
                if (isDirectory) 
                {
                    if (errorCode == Interop.Errors.ERROR_ACCESS_DENIED && ignoreAccessDenied)
                        return;
                        
                    // File not found doesn't make much sense coming from a directory.
                    if (errorCode == Interop.Errors.ERROR_FILE_NOT_FOUND)
                        errorCode = Interop.Errors.ERROR_PATH_NOT_FOUND;
                }
```
nit, remove duplicated check

Reply:
I don't like this suggestion as it increases code nesting